src/coprocessor/codec/data_type/vector.rs

```diff
@@ -443,6 +443,160 @@ impl VectorValue {
 
         Ok(())
     }
+
+    /// Returns maximum encoded size in binary format.
+    pub fn maximum_encoded_size(&self) -> Result<usize> {
+        match self {
+            VectorValue::Int(ref vec) => Ok(vec.len() * 9),
+
+            // Some elements might be NULLs which encoded size is 1 byte. However it's fine because
+            // this function only calculates a maximum encoded size (for constructing buffers), not
+            // actual encoded size.
+            VectorValue::Real(ref vec) => Ok(vec.len() * 9),
+            VectorValue::Decimal(ref vec) => {
```

Comment:
In chunk, we process `Decimal` as a fixed length struct, maybe we could process it in a similar way? And when we support return data encoded by chunk, would we change this function?

Reply:
> In chunk, we process `Decimal` as a fixed length struct, maybe we could process it in a similar way?

It depends on the format itself. However we can assign a maximum size (instead of an approximate max size) for it so that the size can be just `vec.len() * N` which is very fast to calculate.

> And when we support return data encoded by chunk, would we change this function?

Yes. We need to change both `maximum_encoded_size()` and `encode()`. However I prefer to directly reuse encoding functions in chunk when we can return data in chunk.